nixos/modules/testing/test-instrumentation.nix

```diff
@@ -131,6 +131,19 @@ in
     networking.defaultGateway = mkOverride 150 "";
     networking.nameservers = mkOverride 150 [ ];
 
+    # HACK: as of the writing of this hack, many tests that should depend on
+    # network-online.target depend on multi-user.target instead. These tests
+    # should have their dependencies corrected, but until then, make
+    # multi-user.target pull that in.
+    #
+    # Previously this hack was applied under all configurations and broke
+    # things; see:
+    # - https://github.com/systemd/systemd/issues/27822
+    # - https://github.com/systemd/systemd/issues/29388
+    # - https://github.com/NixOS/nixpkgs/issues/180175
+    # - https://github.com/NixOS/nixpkgs/commit/15d761a525a025de0680b62e8ab79a9d183f313d#commitcomment-128564097
+    systemd.targets.network-online.wantedBy = [ "multi-user.target" ];
```

Comment:
Can we fix these tests properly? I think this hack has been going on for too long already.

Reply:
Do I understand correctly that this is just about replacing a bunch of `wait_for_unit("multi-user.target")` with `wait_for_unit("network-online.target")`?

Reply:
Probably. I don't have the machine time to go and run the entire testsuite and find out which ones are sketchy. It's also possible that some services themselves have their deps mis-declared. Overall this is a big can o worms.

I'm actually feeling quite dubious about the idea of patching the tests with a bad hack, because it means that the tests won't catch the bugs, our users perhaps will.

Reply:
Exactly: https://github.com/NixOS/nixpkgs/pull/86273 mentions services breaking, so this PR is already likely to cause a breakage already. Artificially keeping the tests up but breaking the services may be even worse.

Reply:
Hmmm. I guess we will have to figure out who to poke to get some Hydra resources to run all the tests. Actually this is silly. Let's split the dependency change part off this PR, ship the bug fix, and make a new PR to explicitly work through the fallout of the dependency change. I will try to get that done tomorrow.

Reply:
@rnhmjoj I don't think we should block this PR on fixing the tests. That could be a significant effort, and it's not relevant to what this PR is ultimate trying to do. I'd rather get this long-needed bugfix in without having to stall on tests that are already working ok (if a bit wastefully)

Reply:
I think we should leave in the false dependency here because it's not explicitly breaking anything and fix the false dependency by itself in a second pr.

Reply:
Agreed with everything said.